packages/sync/legacy/class.jetpack-sync-actions.php

```diff
@@ -113,7 +114,9 @@ static function sync_allowed() {
 		if ( Jetpack::is_staging_site() ) {
 			return false;
 		}
-		if ( ! Jetpack::is_active() ) {
+
+		$connection_manager = new Connection_Manager();
```

Comment:
Instead of instantiating the object here, can we pass what we have in the Jetpack to the sync in the constructor? This sort of defeats the purpose of having the connection object at all.

Reply:
Makes sense to me!

That required some refactor, since sync doesn't have a proper main instance to hold stuff. So, I've updated the main sync class to a singleton in https://github.com/Automattic/jetpack/pull/12665/commits/db22d6f9e30c65fadcc671ab908f74d763e1cd2a then placed the connection manager object as a dependency injection in https://github.com/Automattic/jetpack/pull/12665/commits/3dc59c66f790ae650383e9282e45aa3b28b11899 and finally in https://github.com/Automattic/jetpack/pull/12665/commits/322353a2522ba779d1518e0f21ca04bf3e2f786d updated the consumers to use that dependency.